### PR TITLE
Provide documentation for String http_escape() method

### PR DESCRIPTION
**Pull request by larpon**
As title states

### CODE REVIEW

File: doc/classes/String.xml [hunk 1]
```diff
@@ -392,6 +392,10 @@
 			<return type="String">
 			</return>
 			<description>
+				Escapes (encodes) a string to URL friendly format. Also referred to as 'URL encode'.
+				[codeblock]
+				print( "https://example.org/?escaped="+"Godot Engine:'docs'".http_escape() )
```
Comment (akien-mga) [inline doc/classes/String.xml]: There shouldn't be spaces after `(` and before `)` as per the GDScript style guide. And there should be spaces around operators like `+`.

I'm not sure the concatenation adds much to the example though, so I'd simplify it by using `https://example.org/?escaped=Godot Engine:'docs'"` directly.

Comment (Larpon) [reply doc/classes/String.xml]: The concatenation is so it's not the complete URL that's encoded - only the part you'd encode in a real URL.
I'll fix the changes and try and see if I can get the squash to work this time

**Comment by Larpon** [comment]
Closing and grouping with `http_escape`

File: doc/classes/String.xml (hunk 1 above)
Comment (akien-mga) [reply doc/classes/String.xml]: Ah right, good point.

**Pull request closed without merging**